src/main/java/me/zeroeightsix/kami/util/graphics/ProjectionUtils.kt

```diff
@@ -0,0 +1,87 @@
+package me.zeroeightsix.kami.util.graphics
+
+import me.zeroeightsix.kami.util.KamiTessellator
+import me.zeroeightsix.kami.util.Wrapper
+import net.minecraft.client.gui.ScaledResolution
+import net.minecraft.client.renderer.ActiveRenderInfo
+import net.minecraft.client.renderer.GLAllocation
+import net.minecraft.util.math.Vec3d
+import org.lwjgl.opengl.GL11.*
+import org.lwjgl.util.vector.Matrix4f
+import org.lwjgl.util.vector.Vector4f
+
+/**
+ * This file is adapted from forgehax/util/math/VectorUtils.java which is licensed under MIT.
+ *
+ * You can find a copy of the original license here:
+ * https://github.com/fr1kin/ForgeHax/blob/master/src/main/java/com/matt/forgehax/util/math/VectorUtils.java
```

Comment:
Change this url to `https://github.com/fr1kin/ForgeHax/blob/2011740/src/main/java/com/matt/forgehax/util/math/VectorUtils.java`

Even though forgehax is unlikely to ever update, it would be nice to avoid weird issues in the future